### PR TITLE
test: Cypress -  added tags - @tag.Git, @tag.JS, @tag.Binding, @tag.Datasource, @tag.ImportExport

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
Renamed testcases starting with Bug to appropriate Feature_Bug eg DS_Bug, JS_Bug etc. and tagged them.

Reverted the airgap change made to 
1. cypress/e2e/Regression/ClientSide/AdminSettings/Admin_settings_spec.js 
2. cypress/e2e/Regression/ClientSide/Autocomplete/Bugs_AC_Spec.ts

Run Binding test cases -` /ok-to-test tags="@tag. Binding"`
Run JS test cases -   `/ok-to-test tags="@tag.JS"`
Run ImportExport test cases -  ` /ok-to-test tags="@tag.ImportExport"`
Run Binding test cases -   `/ok-to-test tags="@tag.Binding"`
Few more datasource testcases added -  `/ok-to-test tags="@tag.Datasource"`
To run both Git and Datasource tagged testcases -` /ok-to-test tags="@tag.Datasource,@tag.Git"`

>
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- [x] Cypress
>
>

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Tests**
  - Enhanced test suite descriptions with relevant tags for better categorization and filtering.
  - Added new test cases for various bug verifications, ensuring features like datasource interactions, API executions, and widget bindings work as expected.
  - Improved existing test cases with additional steps and assertions to cover more scenarios and edge cases.

- **Bug Fixes**
  - Implemented test cases to verify fixes for specific bugs related to data source configurations, JavaScript object parsing, and UI component behaviors.

- **Documentation**
  - Updated test descriptions to provide clearer context and objectives for each test suite.

### CODE REVIEW

**Review by mohanarpit — APPROVED** [review]

**Comment by sharanya-appsmith** [comment]
Note: Need to accomodate the changes of running airgap using cypress grep. Holding off merging this PR until then.

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug22281_WelcomeTour_spec.ts [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
-      _.homePage.CreateNewWorkspace("GuidedtourWorkspace" + uid);
-      _.homePage.CreateAppInWorkspace(
-        "GuidedtourWorkspace" + uid,
-        `GuidedtourApp${uid}`,
-      );
-      _.homePage.NavigateToHome();
+describe(
+  "excludeForAirgap",
+  "Welcome tour spec",
+  { tags: ["@tag.excludeForAirgap"] },
```
Comment (mohanarpit) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug22281_WelcomeTour_spec.ts:6]: @sharanya-appsmith Will this work? This has both types of tags.

**Review by mohanarpit — APPROVED** [review]

**Comment by mohanarpit** [comment]
@sharanya-appsmith I'm approving the PR with the assumption that my comment is going to be resolved without any changes to the code base.

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug22281_WelcomeTour_spec.ts (hunk 1 above)
Comment (sharanya-appsmith) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug22281_WelcomeTour_spec.ts:6]: { tags: ["@tag.excludeForAirgap"] } will not work but excludeForAirgap with tagify works.

**Pull request merged**